### PR TITLE
Add option --skip-passed to skip passed reports

### PR DESCRIPTION
**Pull request by perlpunk**
Issue: https://progress.opensuse.org/issues/92957

### CODE REVIEW

**Comment by perlpunk** [comment]
I don't know why the 3.6 test is failing. Some problem with flake8

**Comment by perlpunk** [comment]
```
ERROR: InvocationError for command /home/travis/build/os-autoinst/openqa_review/.tox/flake8/bin/flake8 (exited with code 1)
```
Oh, apparently an "InvocationError" for flake8 doesn't mean that there was a problem with flake8. It simply means that the style check did not pass 😂

CI is passing now

File: tests/test_openqa_review.py [hunk 1]
```diff
@@ -439,6 +440,17 @@ def bugrefs_test_args_factory():
     return args
 
 
+def test_skip_passed():
+    args = cache_test_args_factory()
+    args.skip_passed = True
```
Comment (okurz) [inline tests/test_openqa_review.py:445]: I suggest to trim down the report a bit by also adding `args.show_empty = False` here. Can you do that?

File: tests/test_openqa_review.py [hunk 2]
```diff
@@ -439,6 +440,17 @@ def bugrefs_test_args_factory():
     return args
 
 
+def test_skip_passed():
```
Comment (okurz) [inline tests/test_openqa_review.py:443]: you are adding this test directly after the function `bugrefs_test_args_factory` but don't use it. Can you just move it above to line 412?

**Review by okurz — CHANGES_REQUESTED** [review]
This looks very good already. I suggest you try to trim down the test data a bit to have less noisy content in the HTML (and json) files. Can you try to manually delete some sections, e.g. header and footer and see if tests still pass? Also likely a single passed and a single failed openQA job should be enough to prove the concept

File: tests/test_openqa_review.py (hunk 2 above)
Comment (perlpunk) [reply tests/test_openqa_review.py]: done

File: tests/test_openqa_review.py (hunk 1 above)
Comment (perlpunk) [reply tests/test_openqa_review.py]: done

**Comment by perlpunk** [comment]
Here is a comparison of a call without and with `--skip-passed`:
```
openqa-review \
-J https://openqa.opensuse.org/group_overview/68,https://openqa.opensuse.org/group_overview/4\
 --include-softfails --no-empty-sections | markdown
```
openqa-review-all
```
openqa-review \
-J https://openqa.opensuse.org/group_overview/68,https://openqa.opensuse.org/group_overview/4\
 --include-softfails --no-empty-sections | markdown
```
openqa-review-skip-passed

I think that shows why I chose those two.

**Comment by perlpunk** [comment]
> This looks very good already. I suggest you try to trim down the test data a bit to have less noisy content in the HTML (and json) files.

I deleted some HTML sections.

Not sure what to delete from JSON.
Should I sit down and try and delete things and re-add it again if the test fails? That might take a while.

**Review by okurz — APPROVED** [review]
Nice. All good. The JSON documents don't need further trimming, was merely "nice to have"

**Pull request merged**